elephant/change_point_detection.py

```diff
@@ -1,48 +1,42 @@
-Essentially, a det of  two-sided window of width `h` (`_filter(t, h, spk)`)
-slides over the spike train within the time `[h, t_final-h]`. This generates a
-`_filter_process(time_step, h, spk)` that assigns at each time `t` the
-difference between a spike lying in the right and left window. If at any time
-`t` this difference is large 'enough' is assumed the presence of a rate Change
-Point in a neighborhood of `t`. A threshold `test_quantile` for the maximum of
-the filter_process (max difference of spike count between the left and right
-window) is derived based on asymptotic considerations. The procedure is
-repeated for an arbitrary set of windows, with different size `h`.
+Change point detection algorithm :cite:`cpd-Messer2014_2027` determines if a
+spike train `spiketrain` can be considered as a stationary process (constant
+firing rate) or not as stationary process (i.e. presence of one or more points
+at which the rate increases or decreases). In case of non-stationarity, the
+output is a list of detected Change Points (CPs).
+
+Essentially, a det of two-sided window of width `h`
+(`_filter(t, h, spiketrain)`) slides over the spike train within the time
+`[h, t_final-h]`. This generates a `_filter_process(time_step, h, spiketrain)`
+that assigns at each time `t` the difference between a spike lying in the right
+and left windows. If at any time `t` this difference is large 'enough', the
+presence of a rate Change Point in a neighborhood of `t` is assumed. A
+threshold `test_quantile` for the maximum of the filter_process (max
+difference of spike count between the left and right windows) is derived based
+on asymptotic considerations. The procedure is repeated for an arbitrary set of
+windows with different sizes `h`.
+
+.. autosummary::
+    :toctree: _toctree/change_point_detection
+
+    multiple_filter_test
+    empirical_parameters
 
 Examples
 --------
-The following applies multiple_filter_test to a spike trains.
-
-    >>> import quantities as pq
-    >>> import neo
-    >>> from elephant.change_point_detection import multiple_filter_test
-    ...
-    >>> test_array = [1.1,1.2,1.4,   1.6,1.7,1.75,1.8,1.85,1.9,1.95]
-    >>> st = neo.SpikeTrain(test_array, units='s', t_stop = 2.1)
-    >>> window_size = [0.5]*pq.s
-    >>> t_fin = 2.1*pq.s
-    >>> alpha = 5.0
-    >>> num_surrogates = 10000
-    >>> change_points = multiple_filter_test(window_size, st, t_fin, alpha,
-    ...                 num_surrogates, time_step = 0.5*pq.s)
-
-References
-----------
-Messer, M., Kirchner, M., Schiemann, J., Roeper, J., Neininger, R., &
-Schneider, G. (2014). A multiple filter test for the detection of rate changes
-in renewal processes with varying variance. The Annals of Applied Statistics,
-8(4),2027-2067.
+>>> import quantities as pq
+>>> from elephant.change_point_detection import multiple_filter_test
+>>> spike_times = [1.1, 1.2, 1.4, 1.6, 1.7, 1.75, 1.8, 1.9, 1.95] * pq.s
```

Comment:
I'm not sure if it's not cleaner and best practice to leave the SpikeTrain object of the original. After all, not having the spike train will make the "end" of the spike train undefined -- which could well be relevant for the algorithm.

Reply:
I've not changed anything in the example - I've just reformatted the lines.
What do you mean?